services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableIndexManagerImpl.java

```diff
@@ -109,9 +116,37 @@ public Void doInTransaction(TransactionStatus status) {
 							// set the new max version for the index
 							tableIndexDao.setMaxCurrentCompleteVersionForTable(
 									tableId, changeSetVersionNumber);
+
+							//once all changes to main table are applied, populate the list-type columns with the changes.
+							for(Map.Entry<ColumnModel, Set<Long>> entry : listColumnsToRowIdMap.entrySet()){
+								tableIndexDao.populateListColumnIndexTable(tableId, entry.getKey(), entry.getValue());
+							}
 							return null;
 						}
 					});
+
+
+		}
+	}
+
+	public static void recordListColumnChanges(Map<ColumnModel, Set<Long>> listColumnsToRowIdMap, Grouping grouping){

```

Comment:
can add a new type of grouping to SparseChangeSet, then iterate over new grouping and both delete then add for each group of rows for each list column.